Campus Density/DiningCells/MealFilterCell.swift

```diff
@@ -45,18 +46,27 @@ class MealFilterCell: UICollectionViewCell {
     }
 
     func setupConstraints() {
-        let padding: CGFloat = Constants.smallPadding
         var index: Int = 0
-        var buttonLeftOffset: CGFloat = padding
-        filterButtons.forEach { button in
-            let buttonWidth = mealLabel(meal: mealModel.meals[index]).widthWithConstrainedHeight(frame.height, font: .sixteen) + labelHorizontalPadding * 2
-            button.snp.makeConstraints({ make in
+        var buttonLeftOffset: CGFloat = 0
+        let numOfMeals: CGFloat = CGFloat(filterButtons.count)
+        let buttonWidth = frame.width / numOfMeals
+        let sliderHeight = buttonHeight
+        for (meal, button) in zip(mealModel.meals, filterButtons) {
+            let buttonWidth = buttonWidth
+            button.snp.makeConstraints { make in
                 make.width.equalTo(buttonWidth)
                 make.height.equalTo(buttonHeight)
                 make.left.equalToSuperview().offset(buttonLeftOffset)
-            })
+            }
+            if meal == mealModel.selectedMeal {
+                lineView = UIView(frame: CGRect(x: CGFloat(index)*buttonWidth, y: sliderHeight, width: buttonWidth, height: 2))
+                lineView.backgroundColor = .densityGreen
+                lineView.layer.borderColor = UIColor.densityGreen.cgColor
+                lineView.layer.borderWidth = 1
+                contentView.addSubview(lineView)
+            }
```

Comment:
Would it be better to make the `lineView` in the configure function or somewhere else and set its location in setupConstraints? Also I don't think the `index` variable is being used anymore so we can delete it

Reply:
Putting the lineView in the configure function is more idiomatic but I ran into some issues with the changing of the position since it required an extra loop variable. This was cleaner since we are changing the position of the slider every time the user clicks on the meal options. Also, index variable is being used to set the x axis of the slider.

Reply:
Ah yes thanks for pointing that out, I guess I didn't see it the first time. Is `CGFloat(index)*buttonWidth` equivalent to `buttonLeftOffset`?

Reply:
Yup, fixed it, thanks for pointing that out